app/discord_relay.py

```diff
@@ -37,28 +54,95 @@ def handle_message(self, message):
+
+                #create a discord file from the email attachment payload
+                discordfile = discord.File(io.BytesIO(file_obj),filename=filename)
+                attachments.append(discordfile)
+        #for attachment in message.iter_attachments():
+
+        try:
+            channelid = self.get_discord_channel(message.get('to'),
+                                message.get('from'),
+                                message.get('subject'))
+        except Exception as ex:
+            print("Error retrieving channel ID using default: " + self.defaultchannelid)
+
+        self.notify_discord_bot(channelid,
+                            message.get('to'),
                             message.get('from'),
                             message.get('subject'),
-                            msg_body)
-
-    def notify_discord(self, to_addr, from_addr, subject, body):
-        webhook_data = { 
-            "embeds": [
-                {
-                    "title": subject,
-                    "description": "",
-                    "fields": [
-                        self.discord_field("To",to_addr),
-                        self.discord_field("From",from_addr),
-                        self.discord_field("Subject",subject),
-                        self.discord_field("Body",body)
-                    ]
-                }
-            ]
-        }
-
-        r=requests.post(self.webhook_url,json=webhook_data)
+                            msg_body,
+                            attachments)
+        
+    #def handle_message(self, message):
+
+
+    def notify_discord_bot(self, channelid,to_addr, from_addr, subject, body,attachments = None):
+        # Set the properties fo the bot client to values that will be sent to discord.
+        print("Generating email at %s" % datetime.datetime.now())
+        print(f'Email Subject: "{subject}"')
+        self.client.channelid = channelid
+        self.client.subject = subject
+        self.client.embeds = discord.Embed(title=subject,description="desc")
+        self.client.embeds.add_field(name="To",value=to_addr,inline=False)
+        print(f'Email To: "{to_addr}"')
+        self.client.embeds.add_field(name="From",value=from_addr)
+        print(f'Email From: "{from_addr}"')
+        self.client.embeds.add_field(name="Subject",value=subject)
+        if body is not None:
+            chunklength = 1000
+            chunks = [body[i:i+chunklength] for i in range(0,len(body),chunklength)]
+            print('Email Body length: %s = %s chunks' % (len(body),len(chunks)))
+            chunknum = 1
+            for chunk in chunks:
+                self.client.embeds.add_field(name="Body-%s" %chunknum,value=chunk)
+                chunknum += 1
+        
+        print('Adding Attachments: %s' % len(attachments) )
+        self.client.files = attachments
+        
+        print('Sending email')
+        #reset flag
+        self.client.msg_sent = False
```

Comment:
This one's a bit of a bigger comment, but I don't really like how this tells the discord bot to send the message. It looks like this is setting some fields in the bot object, then turning on the `msg_sent` flag which the bot looks for in a loop. But I suspect that will have concurrency issues (like if multiple messages come in at once).

I think a better way would be to send an event, passing all of the information as parameters. It looks relatively easy to send custom events:
https://stackoverflow.com/questions/64810905/how-do-i-emit-custom-discord-py-events

Then instead of a timer, you could simply register that event and send the message when it fires.

I'm envisioning something along the lines of
```
embeds  = discord.Embed(title=subject,description="desc")
embeds.add_field(name="To",value=to_addr,inline=False)
...
print('Sending email')
await client.emit('send_email', channelid, subject, embeds, attachments)
```

Then in the bot, instead of
```
    @tasks.loop(seconds=1)
    async def timer(self):
        try:
            if not self.msg_sent:
                print("Email content passed to bot for processing.")
                ...
```

it would be like:
```
async def on_send_email(channelid, subject, embeds, attachments):
    print("Email content passed to bot for processing.")
    # all the stuff for sending the message to the channel
    ...
```

Is this something you'd want to try doing? If not, I can accept this as-is and give it a shot.